hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -1244,15 +1240,19 @@ private OzoneFileStatus getOzoneFileStatus(String volumeName,
 
       // if the key is a file then do refresh pipeline info in OM by asking SCM
       if (fileKeyInfo != null) {
-        if (latestLocationVersion) {
-          slimLocationVersion(fileKeyInfo);
-        }
-        // refreshPipeline flag check has been removed as part of
-        // https://issues.apache.org/jira/browse/HDDS-3658.
-        // Please refer this jira for more details.
-        refresh(fileKeyInfo);
-        if (sortDatanodes) {
-          sortDatanodes(clientAddress, fileKeyInfo);
+        // If operation is head, do not perform any additional steps
+        // As head operation does not need any of those details.
+        if (!args.isHeadOp()) {
+          if (args.getLatestVersionLocation()) {
```

Comment:
Thanks @adoroszlai for taking care this case. 
I think, need to move latest version location outside `if (!args.isHeadOp())` block.
```
        if (args.getLatestVersionLocation()) {
          slimLocationVersion(fileKeyInfo);
        }
```

Reply:
Thanks @rakeshadr for the review.  I have updated this.